browser/src/react-redux/App/SignupPage/SignupPage.react.js

```diff
@@ -24,145 +24,169 @@ class SignupPage extends Component {
-    )
-  }
+    );
+  };
 
   isSubmitButtonDisabled = () =>
     !(
       this.state.email &&
       this.state.password &&
       this.state.confirmPassword &&
       this.state.password === this.state.confirmPassword
-    )
+    );
 
   onSubmit = e => {
-    e.preventDefault()
-    const { emailError, passwordError, generalError } = this.state
+    e.preventDefault();
+    const { emailError, passwordError, generalError } = this.state;
 
     if (emailError || passwordError || generalError) {
       this.setState({
         emailError: '',
         passwordError: '',
         generalError: ''
-      })
+      });
     }
     const user = {
       email: this.state.email,
       password: this.state.password
-    }
+    };
 
     axios
       .post('/api/user', user)
       .then(res => {
         // set current user with redux store.
-        this.props.setUser(res.data.user)
+        this.props.setUser(res.data.user);
       })
       .catch(err => {
         if (err.response.data.error.errors) {
-          const errors = err.response.data.error.errors
+          const errors = err.response.data.error.errors;
 
           errors.email
             ? this.setState({ emailError: errors.email.message })
-            : null
+            : null;
           errors.password
             ? this.setState({ passwordError: errors.password.message })
-            : null
+            : null;
         } else {
           this.setState({
             generalError: err.response.data.error.message
-          })
+          });
         }
-      })
-  }
+      });
+  };
 
   render() {
-    const { user } = this.props
+    const { user } = this.props;
     if (user) {
-      return <Redirect to="/" />
+      return <Redirect to='/' />;
     }
     return (
-      <div className="signup-page">
-        <form onSubmit={this.onSubmit} className="signup-page__form">
-          <h1 className="signup-page__title">Signup for Site Builder</h1>
-          <p className="signup-page__general-error">
+      <div className='signup-page'>
+        <form onSubmit={this.onSubmit} className='signup-page__form'>
+          <h1 className='signup-page__title'>Signup for Site Builder</h1>
+          <p className='signup-page__general-error'>
             {this.state.generalError}
           </p>
 
-          <div className="signup-page__input">
-            <label htmlFor="email" className="signup-page__label">
+          <div className='signup-page__input'>
+            <label className='signup-page__label'>
               Email
-              <span className="signup-page__error">
+              <span className='signup-page__error'>
                 {this.state.emailError}
               </span>
             </label>
+            <input
+              name='email'
+              value={this.state.email}
+              type='email'
+              onChange={this.handleChange}
+              className='signup-page__input-field'
+              required
+            />
```

Comment:
Not sure what happened here, but all the `inputs` got deleted. I had to add them back.